### PR TITLE
Contact snippet fixes and ContactExpandableGroup refactor

### PR DESCRIPTION
**Pull request by chosak**
This PR makes several fixes to how the Contact snippet is rendered and also reimplements how contacts can be rendered as Expandables and in an ExpandableGroup.

It reverts changes introduced in #5334 (Add new ContactExpandableGroup block to LearnPages), which to date have not been deployed anywhere.

See internal el-camino#21 for broader context around this work, intended to reconcile our [Contact Us](https://www.consumerfinance.gov/about-us/contact-us/) page with Contact snippets. This fixes platform#3589 (Bug: ContactEmail molecule ignores specified link text) and platform#3598 (Add URL support to Contact snippet).

Essentially this PR introduces a new molecule (ContactExpandable) for rendering Contact snippets as expandables. The expandable heading comes from the Contact title, and its content comes from  the Contact info (the body is purposefully excluded). A new organism (ContactExpandableGroup) is added as a method for placing a series of these on a page.

## Screenshots

As in #5334, the ContactExpandableGroup shows up as a separate option in the LearnPage content area. 

image

This is what it looks like when you're editing one to add Contact snippets:

image

When editing a Contact snippet, you can now add a Hyperlink:

image

The URL is required, the link text is not:

image

Here's what the ContactHyperlink looks like when rendered (see platform#3598 for discussion on selection of "WEB"):

image

Here's what it looks like when rendered with link text:

image

Here's what it looks like when a Contact snippet has multiple pieces of contact info. Note proper spacing between items, fixing an issue mentioned on el-camino#21:

image

## Notes

@contolini @sonnakim

## Todos

1. @anselmbradford @Scotchester note in the images above, there's an extra space underlined in the hyperlink:

   image

   Is this a known issue, or something due to the way I've written [the template]()? Here's how the HTML is rendered:

   image
2. It'd be better to consolidate this "expandable contact" with the existing Expandable/ExpandableGroup, but that is a larger effort made complicated by Wagtail data migrations. I'd like to revisit that after this is merged in and deployed.

## Checklist

- [x] PR has an informative and human-readable title
- [x] Changes are limited to a single goal (no scope creep)
- [x] Code can be automatically merged (no conflicts)
- [x] Code follows the standards laid out in the [CFPB development guidelines](https://github.com/cfpb/development)
- [x] Passes all existing automated tests
- [x] Any _change_ in functionality is tested
- [x] New functions are documented (with a description, list of inputs, and expected output)
- [ ] Placeholder code is flagged / future todos are captured in comments
- [x] Visually tested in supported browsers and devices (see checklist below :point_down:)
- [ ] Project documentation has been updated
- [x] Reviewers requested with the [Reviewers tool](https://help.github.com/articles/requesting-a-pull-request-review/) :arrow_right:

### CODE REVIEW

**Comment by Scotchester** [comment]
> @anselmbradford @Scotchester note in the images above, there's an extra space underlined in the hyperlink … Is this a known issue, or something due to the way I've written the template?

Hmm... I _think_ the way you have written the template is fine. It seems like it's possibly a problem with the middleware that adds the icon and the `<span class="a-link_text">`. To avoid the underline going under the gap, there can be no whitespace between the last letter of the text and the closing `</span>` tag.

File: cfgov/jinja2/v1/_includes/molecules/contact-email.html [hunk 1]
```diff
@@ -6,12 +6,13 @@
...
-   See [GHE]/flapjack/Modules-V1/wiki/Contact:-Email
+   Create a Contact Email molecule.
 
-   value         Object defined from a StreamField block.
+   value.emails:            An array containing the email addresses.
 
-   value.emails: An array containing the email addresses.
+   email.address:           Email address to link to.
+
+   email.link_text:         Alternate link text (optional).
```
Comment (anselmbradford) [inline cfgov/jinja2/v1/_includes/molecules/contact-email.html]: Looks like this should be `email.text`?

File: cfgov/jinja2/v1/_includes/molecules/contact-hyperlink.html [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+   Create a Contact Hyperlink molecule when given:
+
+   value:                 ContactHyperlink block StreamValue.
+
+   value.url:             Hyperlink URL.
+
+   value.text:            Hyperlink text (optional).
+
+   ========================================================================== #}
+<div class="m-contact m-contact-hyperlink" data-qa-hook="contact-hyperlink">
```
Comment (anselmbradford) [inline cfgov/jinja2/v1/_includes/molecules/contact-hyperlink.html]: I think `data-qa-hook="contact-hyperlink"` is unused.

File: cfgov/v1/atomic_elements/organisms.py [hunk 3]
```diff
@@ -781,73 +780,41 @@ class Expandable(BaseExpandable):
...
-        contacts_by_id = contact_model.objects.in_bulk(contact_ids)
 
-        for block in values:
-            block['contacts'] = [
-                contacts_by_id[contact_id]
-                for contact_id in block['contacts']
-            ]
+class ExpandableGroup(blocks.StructBlock):
+    heading = blocks.CharBlock(required=False)
+    body = blocks.RichTextBlock(required=False)
```
Comment (anselmbradford) [inline cfgov/v1/atomic_elements/organisms.py]: We have to use RichTextBlock here or can it be a regular text block?

File: cfgov/v1/atomic_elements/organisms.py [hunk 4]
```diff
@@ -781,73 +780,41 @@ class Expandable(BaseExpandable):
...
+    is_accordion = blocks.BooleanBlock(required=False)
```
Comment (anselmbradford) [inline cfgov/v1/atomic_elements/organisms.py]: Maybe we should exclude accordion while its status is up in the air, or is it better to have here for completeness?

File: cfgov/v1/atomic_elements/organisms.py [hunk 5]
```diff
@@ -781,73 +780,41 @@ class Expandable(BaseExpandable):
...
+    has_top_rule_line = blocks.BooleanBlock(
+        default=False,
+        required=False,
+        help_text=('Check this to add a horizontal rule line to top of '
+                   'expandable group.')
```
Comment (anselmbradford) [inline cfgov/v1/atomic_elements/organisms.py]: Could add `'the expandable group.'`

File: cfgov/v1/atomic_elements/organisms.py (hunk 3 above)
Comment (chosak) [reply cfgov/v1/atomic_elements/organisms.py]: Git seems confused here. I didn't change the definition of this module, I just moved it. Let me see if I can rebase this or something to fix this up.

File: cfgov/v1/atomic_elements/organisms.py (hunk 4 above)
Comment (chosak) [reply cfgov/v1/atomic_elements/organisms.py]: Same comment as above; I'm not sure why this is showing up as a change.

File: cfgov/jinja2/v1/_includes/molecules/contact-hyperlink.html (hunk 2 above)
Comment (Scotchester) [reply cfgov/jinja2/v1/_includes/molecules/contact-hyperlink.html]: Yeah, the corresponding thing in `contact-email.html` could also be removed.

File: cfgov/v1/atomic_elements/organisms.py (hunk 3 above)
Comment (Scotchester) [reply cfgov/v1/atomic_elements/organisms.py]: Git's not that smart. Moving is changing.

File: cfgov/jinja2/v1/_includes/molecules/contact-email.html [hunk 6]
```diff
@@ -23,7 +24,7 @@
     <p>
         {% for email in value.emails %}
```
Comment (Scotchester) [inline cfgov/jinja2/v1/_includes/molecules/contact-email.html]: Not really related to this PR, but do we really need the ability to have multiple emails within a single ContactEmail when one can just add multiple ContactEmails to a Contact (which is what we do in at least the first example I looked at, Accessibility Feedback)?

Comment (chosak) [reply cfgov/jinja2/v1/_includes/molecules/contact-email.html]: I was wondering about that too. I'd bet the purpose is to allow for the listing of multiple addresses without repeating the "✉️ **EMAIL**" header:

image

Comment (Scotchester) [reply cfgov/jinja2/v1/_includes/molecules/contact-email.html]: Yeah, I guess that is useful. But we should use it, then :)

**Comment by sonnakim** [comment]
Looks great!

File: cfgov/v1/tests/atomic_elements/organisms/test_contact_expandables.py [hunk 7]
```diff
@@ -0,0 +1,138 @@
...
+        html = block.render(block.to_python({
+            'heading': "Contacts",
+            'expandables': [
+                {'contact': pk}
+                for pk in (5, 3, 2, 1, 8)
+            ],
+        }))
+
+        self.assertInHTML("<h3>Contacts</h3>", html)
+        self.assertNotIn("World's richest duck", html)
```
Comment (schbetsy) [inline cfgov/v1/tests/atomic_elements/organisms/test_contact_expandables.py]: Is this assertion meaningful? I might have missed something, but I didn't think this string even appears in any of the `Contact` objects that exist for this test.

File: cfgov/v1/tests/atomic_elements/test_molecules.py [hunk 8]
```diff
@@ -176,10 +174,84 @@ def test_related_metadata(self):
...
+        value = block.to_python({
+            'url': 'https://example.com',
+            'text': 'Example',
+        })
+        self.assertInHTML(
+            '<a href="https://example.com">Example</a>',
+            block.render(value)
+        )
+
+    def test_render_with_link_text(self):
```
[the remark this answers is not in the document]
Comment (chosak) [reply cfgov/v1/tests/atomic_elements/test_molecules.py]: Good catch, thanks. This was a copy-paste mistake from the previous test case. And having the duplicate named test means that the first (real) test wasn't even being executed.

File: cfgov/v1/tests/atomic_elements/organisms/test_contact_expandables.py (hunk 7 above)
Comment (chosak) [reply cfgov/v1/tests/atomic_elements/organisms/test_contact_expandables.py]: You're right, this was another copy/paste mistake, as this is used in the test above. I've added a test message there to clarify the purpose of this line, and removed it here.

**Pull request merged**